COM-88 Implement testing indicator 2

tmvumbi2 opened the pull request:
Jira ticket: https://jembiprojects.jira.com/browse/COM-88

tmvumbi2 committed:
COM-88 Implement testing indicator 2
Implements testing indicator 2

lauravignoli commented on reportssql/report_util_functions.sql:
```diff
@@ -1625,3 +1625,145 @@ BEGIN
 END$$
 DELIMITER ;
+
+-- patientRegisteredBeforeReportStartDate
```
@tmvumbi2 This functions should be added to the testing_report_util_functions.sql file instead.

lauravignoli commented on reportssql/report_util_functions.sql:
```diff
@@ -1625,3 +1625,145 @@ BEGIN
+
+-- patientRegisteredBeforeReportStartDate
+
+DROP FUNCTION IF EXISTS patientRegisteredBeforeReportStartDate;
```
@tmvumbi2 By enforcing the condition that the child has to be registered before the reporting period start date, aren't we excluding the children registered during the reporting period that have a viral test done during the period? As these children won't be counted for the next quarter.

lauravignoli replied:
Richard confirmed this. We can just check that the test date is during the reporting period, as the child needs to be registered to be able to have a test done.

This function can be removed.

lauravignoli commented on reportssql/report_util_functions.sql:
```diff
@@ -1625,3 +1625,145 @@ BEGIN
+
+    SELECT  
+        IF (p_includeStartAge, 
+            timestampdiff(MONTH, p.birthdate, dateAtVirologicHIVTest) BETWEEN p_startAge AND p_endAge, 
```
@tmvumbi2 This is calculating the diff in MONTHS but comparing it to years.
![ageCalcBug](https://user-images.githubusercontent.com/1198827/69325817-72704280-0c53-11ea-9835-f5de786ccda6.png)

lauravignoli replied:
I got confused here, the age is in months. It's working fine.

lauravignoli committed:
COM-88 Remove unnecessary check

By enforcing the condition that the child has to be registered before the reporting period start date, we are excluding the children registered during the reporting period that have a viral test done during the period. As these children won't be counted for the next quarter.

The check is not necessary as the child needs to be registered to be able to have a test done.

lauravignoli committed:
COM-88 Move functions to testing ind file

lauravignoli commented on reportssql/report_util_functions.sql:
```diff
@@ -1625,3 +1625,145 @@ BEGIN
+    SELECT value_datetime INTO result
+    FROM obs o
+    JOIN concept c ON o.concept_id = c.concept_id AND c.retired = 0
+    WHERE voided = 0
```
@tmvumbi2 Here it should be o.voided and o.order_id right?

lauravignoli committed:
COM-88 Fix return statement and update parameter name

Updated the age parameters to say that the age is in months.
